client-side/src/components/settings/Settings.jsx

```diff
@@ -0,0 +1,81 @@
+import React, { useState } from 'react';
+import axios from 'axios';
+import { useTranslation } from 'react-i18next'
+import GenericButton  from '../../stories/Button/GenericButton.jsx';
+import GenericInput from '../../stories/GenericInput/genericInput.jsx'
+
+
+const Setting = (currentUser) => {
+   
+    const preferencesId=currentUser.currentUser.preferences._id;
+    const userId=currentUser.currentUser._id;
+    let sendNotificationTime = currentUser.currentUser.preferences.sendNotificationTime;
+    let  emailFrequency= currentUser.currentUser.preferences.emailFrequency;
```

Comment:
please destructure all fields that you need from currentUser in one line,
something like:
const {userId, sendNotificationTime, emailFrequency ...} = currentUser.currentUser.preferences